tests/FakeItEasy.Specs/IllegalFluentApiSpecs.cs

```diff
@@ -0,0 +1,189 @@
+namespace FakeItEasy.Specs
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using System.Reflection;
+    using FakeItEasy.Tests.TestHelpers;
+    using FluentAssertions;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Xbehave;
+
+    /// <summary>
+    /// Check that certain fluent API syntax constructions are illegal.
+    /// </summary>
+    public static class IllegalFluentApiSpecs
+    {
+        private const string FakeUsageTemplate = @"
+namespace ApiTest
+{{
+    using FakeItEasy;
+    public class Test
+    {{
+        public interface IService
+        {{
+            void AVoidMethod();
+
+            int AnIntMethod();
+
+            int AProperty {{ get; set; }}
+        }}
+
+        public void TestMethod()
+        {{
+            var fake = A.Fake<IService>();
+            {0};
+        }}
+    }}
+}}
+";
+
+        private static IEnumerable<Microsoft.CodeAnalysis.MetadataReference> references = AppDomain.CurrentDomain.GetAssemblies()
+            .Where(a => !a.IsDynamic)
+            .Where(CompilationRequiresAssembly)
+            .Select(a => MetadataReference.CreateFromFile(a.Location))
+            .ToList();
+
+        [Scenario]
+        [UsingCulture("en-US")]
+        [Example(

```

Comment:
It's hard to pick examples. I have included the illegal `Invokes`…`MustHaveHappened`. And some other semi-random stuff. Making an exhaustive list would be exhausting! But if you want more or less of something, I'm happy to change.

Reply:
Wow. You included a *lot* of stuff here. To be honest, I didn't review every example, I'm just going to trust you on this ;)